### PR TITLE
Use "zypper dup --no-vendor-change" equivalent

### PR DESCRIPTION
**Pull request by lslezak**
- Use `Pkg.PkgUpdateAll()` instead of `Pkg.AddUpgradeRepo()`
- Disable the vendor change solver flag
- 3.1.136

### CODE REVIEW

File: src/lib/registration/migration_repositories.rb [hunk 1]
```diff
@@ -17,67 +17,90 @@
...
 
-    # use the repositories from a libzypp service for the migration
-    # @param [String] service_name name of the service
-    def add_service(service_name)
-      service_repos = SwMgmt.service_repos(service_name)
-      repositories.concat(service_repos)
+    # does any configured service contain an update repo?
+    # @return [Boolean] true if at least one service repository is an update
+    #   repository
+    def service_update_repo?
```
Comment (jreidinger) [inline src/lib/registration/migration_repositories.rb]: I miss here some glue betwee service and update repo like `service_with_update_repo?`?

File: src/lib/registration/migration_repositories.rb [hunk 2]
```diff
@@ -17,67 +17,90 @@
...
+      all_repos.each do |repo|
+        repo_data = Yast::Pkg.SourceGeneralData(repo)
+
+        # enabled migration repositories, disable the others, change
+        # the status if it is different than expected
+        next if repositories.include?(repo) == repo_data["enabled"]
+
+        new_state_msg = repo_data["enabled"] ? "Disabling" : "Enabling"
+        log.info "#{new_state_msg} repository #{repo_data["alias"]}"
+        Yast::Pkg.SourceSetEnabled(repo, !repo_data["enabled"])
```
Comment (jreidinger) [inline src/lib/registration/migration_repositories.rb]: why `!` for enabled??? very confusing :confused:

File: src/lib/registration/migration_repositories.rb (hunk 1 above)
Comment (lslezak) [inline src/lib/registration/migration_repositories.rb]: A service contains repositories, this function returns `true` if any service provides an update repository, i.e. at least one update repository is provided by the services.

Comment (jreidinger) [inline src/lib/registration/migration_repositories.rb]: yeah, I get it from code, but method name is not much communicative...if I see below `MigrationRepositories.service_update_repo?` then it is hard to quess what it do

Comment (lslezak) [inline src/lib/registration/migration_repositories.rb]: I see... OK, I'll rename it, `service_with_update_repo?` sounds better.

**Comment by jreidinger** [comment]
LGTM

**Pull request merged**